src/exports/api.ts

```diff
@@ -163,6 +163,82 @@ export class API extends EventEmitter<RooCodeEvents> implements RooCodeAPI {
 		await this.sidebarProvider.postStateToWebview()
 	}
 
+	public async createProfile(name: string): Promise<string> {
+		// Input validation
+		if (!name || !name.trim()) {
+			throw new Error("Profile name cannot be empty")
+		}
+
+		const currentSettings = this.getConfiguration()
+		const profiles = currentSettings.listApiConfigMeta || []
+
+		if (profiles.some((profile) => profile.name === name)) {
```

Comment:
Not a huge deal, but we might want to improve / expand the `ProviderSettingsManager` interface and just proxy these calls through to that object. It already has a few things you could use like fetching a profile by name or id. In general the current implementation seems fine though.